tools/docker/lib/common

```diff
@@ -31,22 +28,94 @@ export CERTBOT_PLUGINS=(
+# understood by `docker build buildx --platform <specification>`.
+# Usage: arch2platform [arm64|arm32v6|arm64v8]
+# If the input is not recognized, an error is returned
+arch2platform() {
     REQUESTED_ARCH="${1}"
-    if [[ "${REQUESTED_ARCH}" == "all" ]]; then
-        ALL_REQUESTED_ARCH=("${ALL_TARGET_ARCH[@]}")
+    case $REQUESTED_ARCH in
+        amd64)
+            echo "linux/amd64"
+            ;;
+        arm32v6)
+            echo "linux/arm/v6"
+            ;;
+        arm64v8)
+            echo "linux/arm64"
+            ;;
+        *)
+            return 1
+            ;;
+    esac
+}
+
+
+# Interpret the user input. Expands "all" to a list of known arches,
+# and validates arches if provided with a list
+InterpretArchRequest() {
+    USER_INPUT="${1}"
+    local IFS=","
+    # Handle the special value "all"
+    if [[ "${USER_INPUT}" == "all" ]]; then
+        # Recursive call using the list of all known architectures cast to
+        # comma separated list
+        echo "${ALL_TARGET_ARCH[*]}"
         return 0
     fi
-    for TARGET_ARCH in "${ALL_TARGET_ARCH[@]}"; do
-        if [[ "${TARGET_ARCH}" == "${REQUESTED_ARCH}" ]]; then
-            ALL_REQUESTED_ARCH=("${REQUESTED_ARCH}")
-            return 0
+    # Convert comma separated list to array of strings
+    read -ra REQUESTED_ARCH_ARRAY <<< "$USER_INPUT"
+    for REQUESTED_ARCH in "${REQUESTED_ARCH_ARRAY[@]}"; do
+        local IFS=" "
+        if [[ ! " ${ALL_TARGET_ARCH[*]} " =~ " ${REQUESTED_ARCH} " ]]; then
+            echo "unknown architecture identifier: ${REQUESTED_ARCH}"
+            exit 1
         fi
     done
-    # If we didn't return above, REQUESTED_ARCH has an unexpected value.
-    echo "Unexpected target architecture \"${REQUESTED_ARCH}\"". >&2
-    exit 1
+    echo "$USER_INPUT"
+}
+
+ParseArgs() {
+
+    export TAG_VER="$1"
+    if [ -z "$TAG_VER" ]; then
+        echo "We cannot tag Docker images with an empty string!" >&2
+        exit 1
+    fi
+    ARCH_LIST="$2"
+    if [ -z "$ARCH_LIST" ]; then
+        echo "Architectures must be specified!" >&2
+        exit 1
+    fi
+    # split arch list into an array for per-arch image building and saving
+    local IFS=","
+    read -ra REQUESTED_ARCH_ARRAY <<< $(InterpretArchRequest "$ARCH_LIST")
+    export REQUESTED_ARCH_ARRAY
+}
+
+
+# Function for use with trap in the primary scripts to remove the 
+# docker builder and restore the original directory
+Cleanup() {
+    docker buildx rm certbot_builder || true
+    popd
+}
+
+# add binfmt tools to the docker environment, with integration into the new builder instance
+InstallMultiarchSupport() {
+    docker run --privileged --rm tonistiigi/binfmt --install all
+}
+
+# Function to create a docker builder using the buildkit docker-container
+# driver
+CreateBuilder() {
+    # just incase the env is not perfectly clean, remove any old instance of the builder
+    docker builder rm certbot_builder || true
```

Comment:
```suggestion
    docker buildx rm certbot_builder || true
```

Reply:
wtf i fixed this in [f01bc93](https://github.com/certbot/certbot/pull/9628/commits/f01bc93f2ce74407d336ef472551553b087654b4)

Reply:
wait no i had that invocation pattern twice